### PR TITLE
Enable the importer to run asynchronously

### PR DESCRIPTION
**Pull request by chessbr**
Refs RM-4

### CODE REVIEW

File: shuup/core/models/_background_tasks.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    class Meta:
+        verbose_name = _("background task")
+        verbose_name_plural = _("background tasks")
+
+
+class BackgroundTaskExecutionStatus(Enum):
+    RUNNING = 0
+    SUCCESS = 1
+    ERROR = 2
```
Comment (ranta) [inline shuup/core/models/_background_tasks.py:48]: Would tasks always be running, or will they also be waiting in queue for other tasks to finish first? In that case we would need a fourth status also

File: shuup/core/tasks.py [hunk 2]
```diff
@@ -38,22 +69,25 @@ def __init__(self, function, **kwargs):
...
+    def create_task(self, function, stored=False, queue="default", **kwargs) -> Task:
         """
         Create a task to run.
 
         :type function: str
         """
         raise NotImplementedError()
 
-    def run_task(self, task):
+    def run_task(self, task) -> Optional[TaskResult]:
```
Comment (ranta) [inline shuup/core/tasks.py:90]: Nice typing of methods ❤️

File: shuup/core/models/_background_tasks.py (hunk 1 above)
Comment (chessbr) [reply shuup/core/models/_background_tasks.py:48]: they will start immediately. the TaskRunner is the one that is responsible for this. But I don't think we need a new state for this.

**Review by ranta — APPROVED** [review]
This is some pretty good stuff 👍

**Pull request merged**